### DISCUSSION TITLE
Volume Fraction

### DISCUSSION THREAD

**Discussion by KOuadie** [mtex-toolbox/mtex#779] [Ask Anything]
Hello,

I'd like to compare between XRD and EBSD data by comparing volume fractions of specific features in the texture. I'd like to plot it as a function of the number of grains.
I'm sure there is a way to code this but I'm a little lost, I tried a few methods I found in the documentation but failed. 

Sincerely,

K.

**Comment by kilir** [comment]
Hi,
a) I'd say getting the number of grains (from grain size and the volume you analyze)  from XRD is  - positively speaking - extremely challenging. 
b) Assuming you, have suitable data, you may have to explain a little more in detail what you want to do.
Cheers,
Rüdiger

**Reply by KOuadie** [reply]
Hi,
You are right. 
Now I'd like to compare the ebsd data done for different planes. Getting the number of grains should be done with the calcGrains function. I'm struggling a bit with the volume fraction part. How can I calculate the volume fraction of different texture component from ODF maps generated from EBSD data?
Thank you for the fast reply.

**Reply by kilir** [reply]
Hi,
have a look at https://mtex-toolbox.github.io/ODFComponents.html#6
Cheers,
Rüdiger

**Reply by KOuadie** [reply]
Thank you very much Rüdiger! I was able to get the volume fractions using the calcComponents function.
I'd like to make a comparison of volume fractions on specific orientations in Euler space ([1 1 0],[2 0 0],[2 1 1],[2 2 0]) and have a plot with the orientations in the x-axis and the volume fractions on y-axis. Is there a code that would allow me to do that?
Thank you for your help.

**Reply by kilir** [reply]
How would you plot an orientation on a 1-dimensional axis? Or do you simply want a categories such as...

![categorial](https://user-images.githubusercontent.com/11697557/108243524-7b014280-714e-11eb-8e09-bd83ef2abe28.png)

Cheers,
Rüdiger

**Reply by KOuadie** [reply]
Yes that's what I mean. The plot I'm trying to reproduce looks like this :
![image](https://user-images.githubusercontent.com/73796229/108243772-c156a180-714e-11eb-9604-d5de7a0a8530.png)

**Reply by kilir** [reply]
Well, that rather looks like plotting densities along a fibre (see here: https://mtex-toolbox.github.io/OrientationFibre.html#14) However, for any list of orientations you can do:
```
% some data
mtexdata dubna
odf = pf.calcODF
cs = pf.CS;
%%

% orientations to evaluate odf 
o = orientation.byEuler([0 30 45; 30 0 15; 70 60 0]*degree,cs)

% some names
names = categorical({'a', 'b', 'c'})

% a plot
bar(names,odf.eval(o))
```
